app/src/main/java/swati4star/createpdf/util/PdfToImages.java

```diff
@@ -26,18 +28,42 @@
     private ExtractImagesListener mExtractImagesListener;
     private int mImagesCount = 0;
     private ArrayList<String> mOutputFilePaths;
+    private PDFEncryptionUtility mPdfEncryptionUtility;
+    private PDFUtils mPDFUtils;
+    private Context mContext;
+    private String mDecryptedPath;
 
-    public PdfToImages(String mPath, Uri mUri, ExtractImagesListener mExtractImagesListener) {
+    public PdfToImages(Context context, String mPath, Uri mUri,
+                       ExtractImagesListener mExtractImagesListener) {
         this.mPath = mPath;
         this.mUri = mUri;
         this.mExtractImagesListener = mExtractImagesListener;
         mOutputFilePaths = new ArrayList<>();
+        this.mContext = context;
+    }
+
+    /**
+     * This function checks whether a file is encrypted or not and if it is
+     * then it decrypts it.
+     *
+     * @param path is the path of the file which needs to be checked for encryption.
+     */
+    private void decryptFile(String path) {
+        if (mPDFUtils.isPDFEncrypted(path)) {
+            mDecryptedPath = mPdfEncryptionUtility.removePasswordForImages(path, null);
+        }
     }
 
     @Override
     protected void onPreExecute() {
         super.onPreExecute();
         mExtractImagesListener.extractionStarted();
+        mPdfEncryptionUtility = new PDFEncryptionUtility((Activity) mContext);
+        mPDFUtils = new PDFUtils((Activity) mContext);
+        if (mUri != null)
```

Comment:
There will not be a case when `mURI` will be null. So every time `URI` is sent to that function and doing only `uri.toString()` we do not get the correct path. So I think the problem will be here.

Reply:
@vyankatesh24 No the mUri is null in most cases but the mPath is never null. Though as you have said I am removing the mUri null case. Please check.

Reply:
I have logged then and have seen.

Reply:
Have you tested the changes?
Is it working?

Reply:
Even after removing it. It has the same kind of problem. Though the password dialog doesn't appear  for non-encrypted files which was the case before also as i said earlier.